test/skimNtuple2016_HHbtag.cpp

```diff
@@ -3583,23 +3583,43 @@ int main (int argc, char** argv)
         TLorentzVector tlv_secondBjet_raw = tlv_secondBjet;
 
         // Total JES up/down variation
-        TLorentzVector tlv_firstBjet_raw_jetupTot    = getShiftedJet(tlv_firstBjet_raw , +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet1idx));
-        TLorentzVector tlv_firstBjet_raw_jetdownTot  = getShiftedJet(tlv_firstBjet_raw , -1., theBigTree.jets_jetUncRegrouped_Total_dw->at(bjet1idx));
-        TLorentzVector tlv_secondBjet_raw_jetupTot   = getShiftedJet(tlv_secondBjet_raw, +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet2idx));
-        TLorentzVector tlv_secondBjet_raw_jetdownTot = getShiftedJet(tlv_secondBjet_raw, -1., theBigTree.jets_jetUncRegrouped_Total_dw->at(bjet2idx));
-        TLorentzVector tlv_bH_raw_jetupTot           = tlv_firstBjet_raw_jetupTot + tlv_secondBjet_raw_jetupTot;
-        TLorentzVector tlv_bH_raw_jetdownTot         = tlv_firstBjet_raw_jetdownTot + tlv_secondBjet_raw_jetdownTot;
+        TLorentzVector tlv_firstBjet_raw_jetupTot   ;    
+        TLorentzVector tlv_firstBjet_raw_jetdownTot ;
+        TLorentzVector tlv_secondBjet_raw_jetupTot  ; 
+        TLorentzVector tlv_secondBjet_raw_jetdownTot; 
+        TLorentzVector tlv_bH_raw_jetupTot	    ;
+        TLorentzVector tlv_bH_raw_jetdownTot	    ; 
 
+        if (isMC)
+	{
+           tlv_firstBjet_raw_jetupTot	 = getShiftedJet(tlv_firstBjet_raw , +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet1idx));
+           tlv_firstBjet_raw_jetdownTot  = getShiftedJet(tlv_firstBjet_raw , -1., theBigTree.jets_jetUncRegrouped_Total_dw->at(bjet1idx));
+           tlv_secondBjet_raw_jetupTot   = getShiftedJet(tlv_secondBjet_raw, +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet2idx));
+           tlv_secondBjet_raw_jetdownTot = getShiftedJet(tlv_secondBjet_raw, -1., theBigTree.jets_jetUncRegrouped_Total_dw->at(bjet2idx));
+           tlv_bH_raw_jetupTot  	 = tlv_firstBjet_raw_jetupTot + tlv_secondBjet_raw_jetupTot;
+           tlv_bH_raw_jetdownTot	 = tlv_firstBjet_raw_jetdownTot + tlv_secondBjet_raw_jetdownTot;
+	}
+	
+	else
+	{
+           tlv_firstBjet_raw_jetupTot	 = tlv_firstBjet_raw ;
+           tlv_firstBjet_raw_jetdownTot  = tlv_firstBjet_raw ;
+           tlv_secondBjet_raw_jetupTot   = tlv_secondBjet_raw;
+           tlv_secondBjet_raw_jetdownTot = tlv_secondBjet_raw;
+           tlv_bH_raw_jetupTot  	 = tlv_firstBjet_raw + tlv_secondBjet;
+           tlv_bH_raw_jetdownTot	 = tlv_firstBjet_raw + tlv_secondBjet;
```

Comment:
just a suggestion to make everything more compact and readable, here and everywhere below:
you could assign the central value when you declare the variable and then overwrite it `if isMC`.
e.g. instead of:
```
TLorentzVector tlv_firstBjet_raw_jetupTot   ;
if (isMC)
  {
      tlv_firstBjet_raw_jetupTot = getShiftedJet(tlv_firstBjet_raw , +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet1idx));
  }
else
  {
     tlv_firstBjet_raw_jetupTot	 = tlv_firstBjet_raw ;
  }
```
do 
```
TLorentzVector tlv_firstBjet_raw_jetupTot =  tlv_firstBjet_raw ;
if (isMC)
  {
      tlv_firstBjet_raw_jetupTot = getShiftedJet(tlv_firstBjet_raw , +1., theBigTree.jets_jetUncRegrouped_Total_up->at(bjet1idx));
  }

Reply:
Thanks for the suggestion! I took it into account!